src/internal/obj/limited_client.go

```diff
@@ -39,17 +52,21 @@ func NewLimitedClient(client Client, maxReaders, maxWriters int) Client {
 }
 
 func (loc *limitedClient) Put(ctx context.Context, name string, r io.Reader) error {
+	t := time.Now()
```

Comment:
This might be more of a general monitoring question, but it's applicable here.  Is it better (or is there a cost) to marking start and end events?  If we run into a problem where everything gets hung up here, the start end delta would make it more obvious.

Reply:
You're right that we should have _started and _finished for this, because waiting until it's done to emit data means that we don't know how many waits are in-flight (or indefinitely blocked).  I'll add those.

I worry a little about the cost around things that aren't IO, but this wraps IO so it doesn't trigger my "obviously a bad idea" heuristic.  And, we have cross-fleet continuous profiling with Cloud Profiler, so if we see that we spend a lot of time in `(*prometheus.Counter).Inc()` then we can just remove or rework the slow metrics.

Reply:
Done.  I didn't add a _finished because the histogram computes that for us.  The `HELP` message explains what that's called, so someone browsing `/metrics` knows that `pachyderm_pfs_object_storage_limited_seconds_count` is the end of `pachyderm_pfs_object_storage_block_started_total`.  I also tested this to be sure it was correct.